Application/src/view/visuals/component/ClassRelationship.java

```diff
@@ -0,0 +1,249 @@
+package view.visuals.component;
+
+import javafx.scene.SnapshotParameters;
+import javafx.scene.canvas.GraphicsContext;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.paint.Color;
+import view.visuals.Renderable;
+
+/**
+ * Class for simulating the relationships of the class diagram
+ *
+ * @author Rashad Kamsheh
+ * @version 1.1
+ */
+
+public class ClassRelationship implements Renderable {
+
+
+    //Image for the road sprite which are used to resemble the class diagram relationships
+    private static Image road = new Image("resources/roadSprite.png");
+    //ImageView that contains the arrow, used ImageView to be able to rotate
+    private static ImageView arrow = new ImageView("resources/road-arrow.png");
+    private Coordinates coordinates; //coordinates
+    private Coordinates fromNode, toNode;
+    private int size; //size of the allowed class space
+
+    // constructor
+    public ClassRelationship(Coordinates fromNode, Coordinates toNode, int size) {
+        this.fromNode = fromNode;
+        this.toNode = toNode;
+        this.size = size;
+    }
+
+    // initialiser for rendering the road sprites between a super class and a sub class
+    public void init(Coordinates fromNode, Coordinates toNode, int size) {
+        this.fromNode = fromNode;
+        this.toNode = toNode;
+        this.size = size;
+    }
+
+    @Override
+    public void render(GraphicsContext gc) {
+
+        int startingPointX = fromNode.getX(); // Points to the middle of the super class, used for horizontal lines
+        int startingPointY = fromNode.getY(); // Points to the middle of the super class, user for vertical lines
+        int endingPointX = toNode.getX(); // Points to the middle of the sub class, used for horizontal lines
+        int endingPointY = toNode.getY(); // Points to the middle of the sub class, used for horizontal lines
+        int XOffset = 0; // Vertical offset
+        int YOffset = 0; // Vertical offset
+        // we need 2 offset increment integers for the case where we have to negate the operand of only one of them
+        int offsetIncrementX = 15; // Increments offset of the X axis offset
+        int offsetIncrementY = 15; // Increments offset of the Y axis offset
+        int XDistance = endingPointX - startingPointX; //if this is negative, ending point it more to the left than starting point on X axis
+        int YDistance = endingPointY - startingPointY; // if this is negative, same as above, for Y axis
+        int gap = 0; // Used to remove steps from the for loops so that the arrow appears correctly
+        int arrowSize = 15; // Initial arrow size is 15 but doubled later for diagonal lines
+        int arrowAngle = 0; // used to rotate the arrow to 8 different directions
+
+        // the following 8 if statements are used to rotate the arrow according to vertical and horizontal distance,
+        // note: looking for a better solution with less complexity
```

Comment:
I googled in like 2 seconds. 
https://www.google.se/search?client=ubuntu&channel=fs&q=get+angle+from+deltax+and+deltay&ie=utf-8&oe=utf-8&gfe_rd=cr&dcr=0&ei=WEwlWq_cKILCwgLYmbd4

First result:
https://stackoverflow.com/questions/15994194/how-to-convert-x-y-coordinates-to-an-angle

Conculsion:
var rad = Math.atan2(deltaY, deltaX);
var deg = rad * (180 / Math.PI);

Reply:
+1

Reply:
@pontuslaestadius i tried that and searched a bit more, not working with the all corner cases bruv

Reply:
What corner cases?